aspnetcore/signalr/diagnostics.md

```diff
@@ -0,0 +1,206 @@
+
+### Other environments
+
+If you are running in another environment (Docker, Kubernetes, Windows Service, etc.), see the full documentation on [ASP.NET Core Logging](xref:fundamentals/logging/index) for more information on how to configure logging providers suitable to your environment.
+
+## JavaScript client logging
+
+> [!NOTE]
+> Client-side logs may contain sensitive information from your application. **Never** post raw logs from production applications to public forums like GitHub.
+
+When using the JavaScript client, you can configure logging options using the `.configureLogging` method on `HubConnectionBuilder`:
+
+[!code-javascript[Configuring logging in the JavaScript client](diagnostics/logging-config-js.js?highlight=3)]
+
+> [!NOTE]
+> To disable logging entirely, specify `signalR.LogLevel.None` in the `configureLogging` method.
+
+Log levels available to the JavaScript client are listed below. Setting the log level to one of these values enables logging of message at that level, as well as all log levels above it on the table below:
+
+| Level | Description |
+| ----- | ----------- |
+| `None` | No messages are logged. |
+| `Critical` | Messages that indicate a failure in the entire app. |
+| `Error` | Messages that indicate a failure in the current operation. |
+| `Warning` | Messages that indicate a non-fatal problem. |
+| `Information` | Informational messages. |
+| `Debug` | Diagnostic messages useful for debugging. |
+| `Trace` | Very detailed diagnostic messages designed for diagnosing specific issues. |
+
+Once you've configured the verbosity, the logs will be written to the Browser Console (or Standard Output in a NodeJS application).
+
+If you want to send logs to a custom logging system, you can provide a JavaScript object implementing the `ILogger` interface. The only method that needs to be implemented is `log`, which takes the level of the event and the message associated with the event. For example:
+
+[!code-typescript[Creating a custom logger](diagnostics/custom-logger.ts?highlight=3-7,13)]
+
+## .NET client logging
+
+> [!NOTE]
+> Client-side logs may contain sensitive information from your application. **Never** post raw logs from production applications to public forums like GitHub.
+
+To get logs from the .NET client, you can use the `.ConfigureLogging` method on `HubConnectionBuilder`. This works the same way as the `.ConfigureLogging` method on `WebHostBuilder` and `HostBuilder`. You can configure the same logging providers you use in ASP.NET Core. However, you have to manually install and enable the NuGet packages for the individual logging providers.
```

Comment:
Nit: method names are referenced in an inconsistent matter. Some, like `.ConfigureLogging` above, start with a period. And others, like `AddConsole` below, don't.